src/devices/DCMotor/DCMotor.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.Gpio;
+using System.Device.Pwm;
+using System.Device.Pwm.Drivers;
+
+namespace Iot.Device.DCMotor
+{
+    public abstract class DCMotor : IDisposable
+    {
+        protected GpioController _controller;
+
+        protected DCMotor(GpioController controller)
+        {
+            _controller = controller ?? new GpioController();
+        }
+
+        public abstract double Speed { get; set; }
+
+        public virtual void Dispose()
+        {
+            _controller?.Dispose();
+            _controller = null;
+        }
+
+        public static DCMotor Create(DCMotorSettings settings)
```

Comment:
Any particular reason why you wanted to go with the factory method instead of using a constructor that gets the settings passed in?

Reply:
There are currently 2 different implementations and implementation depends on the settings (I was wondering if there should be 3 at one point but ended up joining 2 together). I'm almost sure there will be eventually IDCMotor since @Ellerbach's motor implementation abstracted over Spi is a bit similar.

In order to have a constructor I'd have to have a layer of indirection which does not seem necessary since there is very little inconvenience here but a tiny per call perf gain from less indirection

Reply:
I would go more for an interface for those kind of classes than code where we'll pass specific elements (GPIO, SPI, I2C, specific). One of the Reason is that in the case of the implementation I've done, the bellow classes even is thay are using SPI and I2C are specific. Calling a function to move a motor is not the same way. 
Now, once we have our motors with a common interface, it is pretty easy to have higher level calsses like vehicle which can be fully standardized. 
And we can Apply the exact same discussion on Sensors. What should be the interface all sensors should implement so it makes it easier for higher level integration.

Reply:
let's have these conversations separately, I think this is out of scope for this PR